libraries/GCS_MAVLink/GCS_Common.cpp

```diff
@@ -329,11 +330,12 @@ void GCS_MAVLINK::send_ahrs2(AP_AHRS &ahrs)
                                loc.lat,
                                loc.lng);
     }
-    AP_AHRS_NavEKF &_ahrs = reinterpret_cast<AP_AHRS_NavEKF&>(ahrs);
-    if (_ahrs.get_NavEKF2().activeCores() > 0 &&
+    const AP_AHRS_NavEKF &_ahrs = reinterpret_cast<const AP_AHRS_NavEKF&>(ahrs);
```

Comment:
I get that we already had a reinterpret_cast here, but could we move to either making `AP_AHRS_NavEKF` a singleton as well, or move the needed calls into AP_AHRS as overrides and handle the returns being invalid here? (The same way `get_hagl(float)` was moved)

I'm worried about this mostly for future compatibility reasons.